server/src/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -530,6 +534,13 @@ public boolean deleteSnapshot(long snapshotId) {
             return false;
         }
 
+        List<VolumeDetailVO> volumesFromSnapshot;
+        volumesFromSnapshot = _volumeDetailsDaoImpl.findDetails("SNAPSHOT_ID", String.valueOf(snapshotId), null);
+
+        if (volumesFromSnapshot.size() > 0) {
```

Comment:
There could be race condition here. After volumesFromSnapshot = 0, a new volume can be created from snapshot, before the snapshot is actually deleted.